### PR TITLE
Add date class

### PR DESCRIPTION
**Pull request by Sam-Burns**
Import Date class from another project.

### CODE REVIEW

File: lib/Clock.php [hunk 1]
```diff
@@ -19,4 +19,6 @@ public function now(): \DateTimeImmutable;
      * prevent unintended side effects.
      */
     public function nowMutable(): \DateTime;
+
+    public function nowDate(): Date;
```
Comment (ben-challis) [inline lib/Clock.php:23]: Perhaps `today(): Date` as just discussed?

File: lib/Date.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    public function isBetween(self $start, self $end): bool
+    {
+        return !$this->isBefore($start) && !$this->isAfter($end);
+    }
+
+    public function toDateTime(): \DateTimeImmutable
+    {
+        \assert(date_default_timezone_get() === 'UTC', 'System is running with a non-UTC timezone');
```
Comment (ben-challis) [inline lib/Date.php:108]: We'll likely have to remove this I feel as it's rather strict.

**Review by ben-challis — COMMENTED** [review]
Looks good, comment on method naming and the ported assertion being rather strict for this library.

**Review by zoka123 — APPROVED** [review]

File: lib/Date/InvalidDate.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lendable\Clock\Date;
+
+class InvalidDate extends \InvalidArgumentException
```
Comment (ben-challis) [inline lib/Date/InvalidDate.php:7]:
```suggestion
final class InvalidDate extends \InvalidArgumentException
```

nit: internal class

**Review by ben-challis — COMMENTED** [review]
Really minor change here I missed before, LGTM

**Review by ben-challis — APPROVED** [review]

**Pull request merged**